src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,215 @@
+             :divider {:height 1
+                       :margin-horizontal -20
+                       :margin-top (- 12 token-tag-vertical-spacing)
+                       :margin-bottom 8}
+             :membership-request-denied {:margin-top 13}
+             :enter-button {:margin-top 16}
+             :info-text {:margin-top 12
+                         :text-align :center
+                         :padding-horizontal 20}})
+
+(defn multiple-token-requirements? [token-requirement-lists]
+  (when (vector? (first token-requirement-lists))
+    (> (count token-requirement-lists) 1)))
+
+(defn is-token-requirement-met? [token-requirement-list]
+  (and (seq token-requirement-list)
+       (every?
+        (fn [token]
+          (get token :is-sufficient?))
+        token-requirement-list)))
+
+(defn are-multiple-token-requirements-met? [token-requirement-lists]
+  (if (multiple-token-requirements? token-requirement-lists)
+    (some is-token-requirement-met? token-requirement-lists)
+    (is-token-requirement-met? token-requirement-lists)))
+
+(defn token-requirement-list-row [tokens]
+  [rn/view {:style (merge
+                    (get styles :token-row)
+                    (get styles :token-requirement-list-spacing))}
+   (map-indexed (fn [token-index token]
+                  (let [{:keys [token-img-src token amount :is-sufficient? :is-purchasable?]} token]
+                    ^{:key token-index}
+                    [rn/view {:style (get styles :token-tag-spacing)}
+                     [token-tag/token-tag {:token          token
+                                           :value          amount
+                                           :size           24
+                                           :is-required    is-sufficient?
+                                           :is-purchasable is-purchasable?
+                                           :token-img-src  token-img-src}]]))
+                tokens)])
+
+(defn token-requirement-list [props]
+  (let [{:keys [gate]} props
+        [gate-type token-requirement-lists] gate
+        multiple-token-requirements? (multiple-token-requirements? token-requirement-lists)
+        is-sufficient? (are-multiple-token-requirements-met? token-requirement-lists)
+        not-text (if-not is-sufficient? " not" "")
+        message (if
+                 (= gate-type :join)
+                  (i18n/label
+                   :t/community-join-requirements {:not not-text})
+                  (if (= gate-type :read)
+                    (i18n/label
+                     :t/community-channel-read-requirements {:not not-text})
+                    (i18n/label
+                     :t/community-channel-write-requirements {:not not-text})))]
+    [rn/view
+     [rn/view {:style (get styles :token-requirement-text-spacing)}
+      [text/text {:weight :medium
+                  :number-of-lines 1} message]
+      [text/text {:size :paragraph-2
+                  :number-of-lines 1} "You must hodl:"]]
```

Comment:
1. Please move this string to [translations](https://github.com/status-im/status-mobile/blob/develop/translations/en.json) file
2. Maybe designs are changed, but it is not "hodl" anymore. [design](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=2463%3A39587)